docs/design/libraries/LibraryImportGenerator/Compatibility.md

```diff
@@ -54,9 +54,9 @@ In the built-in system, marshalling a `string` contains an optimization for para
 
 When marshalling as [ANSI](https://docs.microsoft.com/windows/win32/intl/code-pages) on Windows (using `UnmanagedType.LPStr`):
   - Best-fit mapping will be disabled and no exception will be thrown for unmappable characters. In the built-in system, this behaviour was configured through [`DllImportAttribute.BestFitMapping`] and [`DllImportAttribute.ThrowOnUnmappableChar`]. The generated marshalling code will have the equivalent behaviour of `BestFitMapping=false` and `ThrowOnUnmappableChar=false`.
-  - No optimization for stack allocation will be performed. Marshalling will always allocate through `AllocCoTaskMem`.
 
 The p/invoke source generator does not provide an equivalent to using `CharSet.Auto` in the built-in system. If platform-dependent behaviour is desired, it is left to the user to define different p/invokes with different marshalling configurations.
+Similarly, `UnmanagedType.LPStr` will only mean ANSI rather than ANSI on Windows and UTF-8 on non-Windows. `UnmanagedType.LPUTF8Str` or `StringMarshalling.Utf8` can be used for UTF-8 and it is left to the user to define different p/invokes if platform-dependent behaviour is desired.
```

Comment:
This is an attempt to make ANSI no longer the same as UTF-8 on non-Windows, so that ANSI is more don't use on non-Windows.

The way I have it right now, for `UnmanagedType.LPStr`, the generator won't do the ANSI on Windows and UTF-8 on non-Windows and `AnsiStringMarshaller` is Windows-specific, so running the generated code on non-Windows will fail. The ANSI means ANSI on Windows and UTF-8 on non-Windows behaviour seemed like some of the odd built-in behaviour we were trying to avoid with the p/invoke generator.

Reply:
> AnsiStringMarshaller is Windows-specific

I do not see anything in the current delta that makes AnsiStringMarshaller Windows specific and fail on non-Windows platforms. Is the split something that remains to be done?

Reply:
I... thought I had it `SupportedOSPlatform("windows")` and using APIs only supported on Windows for some reason - but clearly not. Must be losing my mind.